### PR TITLE
SIMD for Silk.NET.Math #3

### PR DESCRIPTION
**Pull request by WhiteBlackGoose**
## Summary of the PR
Working on #666 

## Tasks

- [x] `IsFinite`
- [x] `IsInfinity`
- [x] `IsNaN`
- [x] `IsNegative`
- [x] `IsNegativeInfinity`
- [x] `IsNormal`
- [x] `IsPositiveInfinity`
- [x] `Sign`

## Tests

- [x] `IsFinite`
- [x] `IsInfinity`
- [x] `IsNaN`
- [x] `IsNegative`
- [x] `IsNegativeInfinity`
- [x] `IsNormal`
- [x] `IsPositiveInfinity`
- [x] `Sign`

#### State of number
- [x] `IsFinite`
- [x] `IsInfinity`
- [x] `IsNaN`
- [x] `IsNegative`
- [x] `IsNegativeInfinity`
- [x] `IsNormal`
- [x] `IsPositiveInfinity`
- [x] `Sign`

#### Other
- [x] Bitwise arithmetics
- [x] Abs

### CODE REVIEW

File: src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs [hunk 1]
```diff
@@ -360,9 +360,9 @@ static bool Decimal(T f)
             [MethodImpl(MaxOpt)]
             static bool SByte(T f)
             {
-                if (typeof(T) == typeof(int))
+                if (typeof(T) == typeof(sbyte))
                 {
-                    return (int) (object) f < 0;
+                    return (sbyte) (object) f < 0;
```
Comment (WhiteBlackGoose) [inline src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs:365]: A bug from scalar

File: src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs [hunk 2]
```diff
@@ -476,7 +487,13 @@ static bool Float(T f)
             {
                 if (typeof(T) == typeof(float))
                 {
-                    return IsNormal((float) (object) f);
```
Comment (WhiteBlackGoose) [inline src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs:479]: A bug from scalar

File: src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs [hunk 3]
```diff
@@ -487,7 +504,13 @@ static bool Double(T f)
             {
                 if (typeof(T) == typeof(double))
                 {
-                    return IsNormal((double) (object) f);
```
Comment (WhiteBlackGoose) [inline src/Maths/Silk.NET.Maths/Scalar.BaseOps.cs:490]: and another bug from scalar

**Review by WhiteBlackGoose — COMMENTED** [review]
There turns out a few bugs in scalar 🦆

**Comment by WhiteBlackGoose** [comment]
#### Benchmark reports
[Silk.NET.Maths.Benchmark.SimdBenchmarks-asm.md](https://github.com/dotnet/Silk.NET/files/7518193/Silk.NET.Maths.Benchmark.SimdBenchmarks-asm.md)
[Silk.NET.Maths.Benchmark.SimdBenchmarks-report-github.md](https://github.com/dotnet/Silk.NET/files/7518195/Silk.NET.Maths.Benchmark.SimdBenchmarks-report-github.md)

Some methods are not inlined. Investigating...

**Review by HurricanKai — APPROVED** [review]

**Pull request merged**